test/ruby/test_yjit.rb

```diff
@@ -1623,6 +1623,40 @@ def entry = self.class
     RUBY
   end
 
+  def test_runtime_stat_floats
```

Comment:
This is something useful to test this PR, but I personally feel this kind of test ends up being a burden in the future, given the fact that it's essentially duplicating the implementation and you'll be forced to replicate the implementation whenever you touch it.

I would test only that those keys are encoded as a Float object by `is_a?(Float)`, or just omit this test.

Reply:
Indeed, that seems wise.  I also compared before and after diffs of the stat output, so I'm confident that the code change has produced no unexpected changes in the result.
Once we merge this, we might decide to change something about the result and wouldn't need this test hindering us at that point 👍